deltachat-ios/Chat/AudioController.swift

```diff
@@ -93,22 +93,29 @@ open class AudioController: NSObject, AVAudioPlayerDelegate, AudioMessageCellDel
             return
         }
         
-        if let fileURL = message.fileURL {
-            let audioAsset = AVURLAsset.init(url: fileURL, options: nil)
-
-            audioAsset.loadValuesAsynchronously(forKeys: ["duration"]) {
-                var error: NSError?
-                let status = audioAsset.statusOfValue(forKey: "duration", error: &error)
-                switch status {
-                case .loaded:
-                    let duration = audioAsset.duration
-                    let durationInSeconds = CMTimeGetSeconds(duration)
-                    DispatchQueue.main.async {
-                        successHandler(messageId, Double(durationInSeconds))
+        DispatchQueue.global(qos: .userInitiated).async {
+            let duration = message.duration
+            if duration > 0 {
+                DispatchQueue.main.async {
+                    successHandler(messageId, Double(duration / 1000))
+                }
+            } else if let fileURL = message.fileURL {
+                let audioAsset = AVURLAsset.init(url: fileURL, options: nil)
+                audioAsset.loadValuesAsynchronously(forKeys: ["duration"]) {
+                    var error: NSError?
+                    let status = audioAsset.statusOfValue(forKey: "duration", error: &error)
+                    switch status {
+                    case .loaded:
+                        let duration = audioAsset.duration
+                        let durationInSeconds = Double(CMTimeGetSeconds(duration))
+                        message.setDuration(1000 * durationInSeconds)
```

Comment:
i assume, the goal of that line is to write the changes to the database for later, faster access?

this does not work that way, `dc_msg_set_*()` methods only change the message object, usually to send it or save it as a draft later.

to actually change the values in the database, use [`dc_msg_latefiling_mediasize()`](https://c.delta.chat/classdc__msg__t.html#a7687ff969841f3d00c3a212f9ad27861)

Reply:
thanks for the hint, I've fixed that